lint/src/main/java/permissions/dispatcher/PermissionsDispatcherIssueRegistry.java

```diff
@@ -4,15 +4,101 @@
 import com.android.tools.lint.detector.api.Issue;
 
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
-@SuppressWarnings("unused")
+import permissions.dispatcher.detectors.PsiCallNeedsPermissionDetector;
+import permissions.dispatcher.detectors.PsiCallOnRequestPermissionsResultDetector;
+import permissions.dispatcher.detectors.PsiNoCorrespondingNeedsPermissionDetector;
+import permissions.dispatcher.detectors.UastCallNeedsPermissionDetector;
+import permissions.dispatcher.detectors.UastCallOnRequestPermissionsResultDetector;
+import permissions.dispatcher.detectors.UastNoCorrespondingNeedsPermissionDetector;
+
 public final class PermissionsDispatcherIssueRegistry extends IssueRegistry {
+
+    static final Provider UAST = new UastImplementation();
+    static final Provider PSI = new PsiImplementation();
+
+    // Providers are asked to provide Detectors in the order that they appear in this list
+    private static final List<Provider> DEFAULT_PROVIDERS = Arrays.asList(UAST, PSI);
+
+    private final List<Provider> providers;
+
+    public PermissionsDispatcherIssueRegistry() {
+        this(DEFAULT_PROVIDERS);
+    }
+
+    PermissionsDispatcherIssueRegistry(List<Provider> providers) {
+        this.providers = providers;
+    }
+
     @Override
     public List<Issue> getIssues() {
-        return Arrays.asList(
-                CallNeedsPermissionDetector.ISSUE,
-                CallOnRequestPermissionsResultDetector.ISSUE,
-                NoCorrespondingNeedsPermissionDetector.ISSUE);
+        for (Provider provider : providers) {
```

Comment:
Nit: for me those lines look a bit awkward because it depends on the order of `DEFAULT_PROVIDERS` and if we change the order it would get messed up right? just checking classpath at first is enough for me👀

Reply:
Yes, `DEFAULT_PROVIDERS` is an ordered List where each element is checked in sequence! There is a comment notifying about the relevance & implications of that at the declaration site of that variable, though.